src/main/scala/is/hail/stats/DownsampleCombiner.scala

```diff
@@ -0,0 +1,127 @@
+package is.hail.stats
+
+import org.apache.spark.sql.Row
+import is.hail.utils._
+
+class DownsampleCombiner(nDivisions: Int) extends Serializable {
+  var binsToCoords: Map[(Int, Int), (Double, Double)] = Map.empty[(Int, Int), (Double, Double)]
+  var initFrame: IndexedSeq[(Double, Double)] = FastIndexedSeq.empty[(Double, Double)]
```

Comment:
Suggestion: I'm not sure what the performance of a FastIndexedSeq is compared to an ArrayBuilder. I'll let @cseed comment on whether it's a problem in this use case.

Reply:
@jigold @cseed this was originally just a List, and it will never get bigger than length = 2